src/components/legend/draw.js

```diff
@@ -25,31 +25,43 @@ var helpers = require('./helpers');
 var MAIN_TITLE = 1;
 
 module.exports = function draw(gd, opts) {
-    if(!opts) opts = gd._fullLayout.legend || {};
-    return _draw(gd, opts);
+    if(opts) {
+        drawOne(gd, opts);
+    } else {
+        var legends = gd._fullLayout._legends;
+        for(var i = 0; i < legends.length; i++) {
+            var legendId = legends[i];
+            var legendObj = gd._fullLayout[legendId];
+            drawOne(gd, legendObj);
```

Comment:
what if you `relayout` or `react` to remove a legend that was previously drawn? Does this work?

Reply:
Both `react` & `relayout` calls appear to work as expected.
Namely it is easy to change the `visible` attribute.
```js
Plotly.relayout(gd, 'legend2.visible', false);
```

Reply:
Try this:
```js
fig1={data:[{y:[1,2,3]}]};
fig2 = {data:[{y:[1,2,3]},{y:[3,1,2],legend:'legend2'}],layout:{legend2:{y:0.5}}};
Plotly.react(gd,fig2)
Plotly.react(gd,fig1)
```
result: legend2 is still there
<img width="759" alt="Screenshot 2023-04-19 at 11 41 25" src="https://user-images.githubusercontent.com/2678795/233128623-2d33e4a3-0cfe-4db0-a71e-377c1d153636.png">

Reply:
Good call. Addressed in https://github.com/plotly/plotly.js/pull/6535/commits/d7f278aba67831646862d6f7df82b7d0335abaa9.